ui/legacy/src/components/cstudio-admin/mods/content-type-propsheet/repository.js

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2007-2020 Crafter Software Corporation. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as published by
+ * the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+(function () {
+
+  const { React, ReactDOM } = CrafterCMSNext;
+  const { useState } = React;
+
+  function Selector({ initialValue, updateFn, rootPath, validations }) {
+    const [value, setValue] = useState(initialValue);
+
+    const onChange = (value) => {
+      setValue(value);
+      updateFn(value);
+    };
+
+    const onBlur = () => {
+      if (!value.startsWith(validations.startsWith)) {
+        if (value.startsWith('/')) {
+          setValue(validations.startsWith + value);
+        } else {
+          setValue(`${validations.startsWith}${value ? '/' + value : value}`);
+        }
+      }
+      updateFn(value);
+    };
+
+    const openPathBrowser = () => {
+      let unmount;
+      const dialogContainer = document.createElement('div');
+      CrafterCMSNext.render(dialogContainer, 'PathBrowserDialog', {
+        open: true,
+        rootPath,
+        initialPath: value.endsWith('/') ? value.slice(0, -1) : value,
+        onClose: () => unmount(),
+        onOk: (path) => {
+          unmount();
+          onChange(path);
+        }
+      }).then((done) => (unmount = done.unmount));
+    };
+
+    return (
+      <>
+        <input
+          type="text"
+          value={value}
+          className="repository-selector--input"
+          placeholder='/site/...'
+          onBlur={validations.startsWith ? onBlur : null}
+          onChange={(e) => onChange(e.target.value)}
+        />
+        <i
+          className="repository-selector--icon fa fa-search"
+          aria-hidden="true"
+          onClick={openPathBrowser}
+        />
```

Comment:
Wrap with a button. Make sure you can tab between input and button in that order.